### ISSUE TITLE
Website verification: add a */* fallback to the Accept request header for improved website compatibility

### ISSUE THREAD

**Issue by luontola** [mastodon/mastodon#22418]
### Steps to reproduce the problem

1. Run a PHP site on Apache HTTP Server
2. Add `Options MultiViews` to your `.htaccess` file, but don't add `MultiviewsMatch Any` (see [this StackOverflow answer](https://stackoverflow.com/questions/16357933/no-acceptable-variant-from-multiviews-in-apache/24598848#24598848) for an explanation)
3. Add your Mastodon verification tag to `https://example.com/page.php`
4. Check with your web browser that you can access the page also at URL `https://example.com/page` in addition to `https://example.com/page.php`
5. In your Mastodon profile, add a link to `https://example.com/page`

### Expected behaviour

The server will return HTTP 200 to Mastodon and website verification succeeds.

### Actual behaviour

The server will return HTTP 406 to Mastodon and website verification fails.

### Detailed description

I had problems verifying my website (see https://github.com/hachyderm/community/issues/110#issuecomment-1356358613 onwards) due to an [Apache httpd configuration mistake](https://stackoverflow.com/questions/16357933/no-acceptable-variant-from-multiviews-in-apache/24598848#24598848) that rarely causes problems, because the web browsers are lenient in their [`Accept` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept). For example Chrome's `Accept` header is something like `text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8`.

But Mastodon's `Accept` header is only `text/html`. This causes the request to fail on websites which for reason or another use a slightly different content type (such as `application/x-httpd-php` in my case).

Please change Mastodon's http client's `Accept` header to for example `text/html,*/*;q=0.8` so that it would be more robust in handling the quirks of various web servers.

### Specifications

Mastodon 4.0.2
Apache HTTP Server 2.4.54
PHP 8.2.0

**Comment by rbairwell** [comment]
I don't think this is a good idea. Mastodon only validates a webpage - in this instance, if */* was allowed, then people could maliciously upload text files or images (with the verification tag as an JPEG comment) which would incorrectly verify.

`application/x-httpd-php` isn't actually a valid/official/recognised [MIME/media type](https://www.iana.org/assignments/media-types/media-types.xhtml), but an Apache handler - the PHP script should return an appropriate Content-Type header (which PHP returns as text/html [by default](https://www.php.net/manual/en/ini.core.php#ini.default-mimetype) ). In this case, it is your server which is misconfigured (let me know if you need a hand fixing it - I'm actually a PHP dev/sysops person - I'm on Hachyderm as the same username).

**Comment by luontola** [comment]
I fixed my server configuration before creating this issue. But since it's a very common configuration error, I doubt that I'm the only one.

Might be a good idea to check the logs of a popular Mastodon server that how often website validations return HTTP 406.

**Comment by luontola** [comment]
> I don't think this is a good idea. Mastodon only validates a webpage - in this instance, if `*/*` was allowed, then people could maliciously upload text files or images (with the verification tag as an JPEG comment) which would incorrectly verify.

Good point about potential attack vectors. Though even then the profile would only show the link to the image - it would validate that you uploaded the image, not that you own the whole domain. Likewise if a site has an XSS vulnerability, the link would show all the query parameters that were used to trigger the XSS so it would be easy to see that the user doesn't own the site.

A more critical issue is making the errors visible to users (see https://github.com/mastodon/mastodon/issues/22417), especially since Mastodon is stricter than the web browsers in what pages it displays.

**Comment by luontola** [comment]
> I don't think this is a good idea. Mastodon only validates a webpage - in this instance, if `*/*` was allowed, then people could maliciously upload text files or images (with the verification tag as an JPEG comment) which would incorrectly verify.

Even if the Accept header is `*/*`, you can still ignore all responses whose Content-Type is not `text/html`. That'll avoid the issue of reading a non-webpage.
